perfkitbenchmarker/providers/ibmcloud/ibm_api.py

```diff
@@ -0,0 +1,577 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities for working with IBM Cloud resources."""
+
+import threading
+import time
+import logging
+import json
+
+from absl import flags
+from perfkitbenchmarker.providers.ibmcloud import ibmcloud
+from perfkitbenchmarker.providers.ibmcloud import util
+
+FLAGS = flags.FLAGS
+
+AVAILABLE = 'available'
+RUNNING = 'running'
+STOPPED = 'stopped'
+
+
+class IbmAPICommand(object):
+  """A IBM Cloud rest api command.
+
+  Attributes:
+  
+    args: list of strings. Positional args to pass, typically
+        specifying an operation to perform (e.g. ['image', 'list'] to list
+        available images).
+        
+    flags: a dict mapping flag name string to flag value. Flags to pass to
+        ibm cloud rest api. If a provided value is True, the flag is passed without a value. 
+        If a provided value is a list, the flag is passed multiple
+        times, once with each value in the list.
+  """
+
+  gen = None
+  gen_instmgr = None
+  gen_imgmgr = None
+  gen_keymgr = None
+  gen_subnetmgr = None
+  gen_fipmgr = None
+  gen_volumemgr = None
+  gen_vpcmgr = None
+  gen_sgmgr = None
+  ibmcloud_account_id = None
+  ibmcloud_apikey = None
+  ibmcloud_auth_token = None
+  ibmcloud_auth_token_time = 0
+  _lock = threading.Lock()
+  base_time = 0.0
+  num_requests = None
+  debug = False
+
+  def __init__(self, resource, *args):
```

Comment:
A lot of times people set the self.flags after using this.  It might be helpful to allow for passing them in here, like this:

```
In [125]: class Foo:
     ...:     def __init__(self, resource, *args, **kwargs):
     ...:         self.resource = resource
     ...:         self.args = args
     ...:         self.flags = kwargs
     ...: 

In [126]: f=Foo('hi', 1, 2, 3, name='joe', date='20201105')

In [127]: f.args
Out[127]: (1, 2, 3)

In [128]: f.flags
Out[128]: {'name': 'joe', 'date': '20201105'}
```

so in this case:
```
 def __init__(self, resource, *args, **kwargs):
   ...
   self.flags = kwargs
```

Reply:
ah ok, I see, I will try that.